File: C/beginner/1080.c
// https://judge.beecrowd.com/pt/problems/view/1080
// Maior e Posição

// Libraries
#include <stdio.h>

// Main Function
int main (){
    // Variables
    int usr_inputs[100];
    int usr_highest, usr_highest_pos;

    // Input
    for (int i = 0; i < 100; i++){
        scanf("%d", &usr_inputs[i]);
    }

    usr_highest = 0;
    usr_highest_pos = 0;

    // Processing

    for (int i = 0; i < 100; i++){
        if (usr_inputs[i] > usr_highest){
            usr_highest = usr_inputs[i];
            usr_highest_pos = i;
        }
    }

    // Output
    printf("%d\n", usr_highest);
    printf("%d\n", usr_highest_pos + 1);

    // Ending Code
    return 0;
}

// By Fernando Serra
// https://github.com/fernandosserra

File: Haskell/beginner/1080.hs
-- https://judge.beecrowd.com/pt/problems/view/1080
-- Maior e Posição

-- Importing ReplicateM
import Control.Monad (replicateM) -- To get sequencial values from user in different lines
import Data.List (elemIndex) -- To get index
import Data.Maybe (fromJust) -- To format indexes output

-- Function to get sequencial values
getValue :: IO Int
getValue = do
    input <- getLine
    let value = read input :: Int
    return value

main :: IO ()
main = do
    values <- replicateM 100 getValue
    let max = maximum values-- To get the maximum values
    let maxIndex = fromJust (elemIndex max values) + 1 -- To Get the index
    print max
    print maxIndex

-- By Fernando Serra
-- https://github.com/fernandosserra